src/storage.rs

```diff
@@ -1630,6 +1630,10 @@ impl Cache {
         }
 
         if cfg.mock_cache {
+            #[cfg(not(test))]
+            let publisher_cache = Default::default();
+            #[cfg(test)]
+            let publisher_cache = crate::tests::mock_publisher_cache();
```

Comment:
FWIW you could make this whole block `#[cfg(test)]` I think - `mock_cache` is test-only.